File: sephora-frontend/src/models/product/EditProductDto.ts
import CharacteristicDto from "../characteristics/CharacteristicDto";

export default interface EditProductDto {
    get id(): number;

    get name(): string | null;

    get descriptionEn(): string;

    get descriptionUa(): string;

    get active(): boolean;

    get brandId(): number;

    get categoryId(): number;

    get characteristics(): CharacteristicDto[];

    get deletedCharacteristics(): number[];
}

File: sephora-backend/Application/Dtos/Product/EditProductDTO.cs
namespace CleanArchitecture.Application.Dtos.Product;

public class EditProductDto
{
    public long Id { get; set; }

    public string Name { get; set; } = String.Empty;

    public string DescriptionEn { get; set; } = String.Empty;

    public string DescriptionUa { get; set; } = String.Empty;

    public bool Active { get; set; }

    public int BrandId { get; set; }

    public int CategoryId { get; set; }

    public IEnumerable<CharacteristicDto> Characteristics { get; set; } = [];

    public IEnumerable<long> DeleteCharacteristics { get; set; } = [];
}
